# Profitable App Profiles for the Google Play Market and Apple App Store

The project is about finding the ideal profiles for an app, in order for it to be profitable. We work for a company that makes apps for both markets, and our jobs is to advise developers so they can make data-driven decisions.

The company only produces free apps, with the main source of revenue being in-app ads. As a result, the revenue is directly related to the number of users; our goal is to analyse data and help developers understand which apps are more popular.

- The [Apple dataset](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home) we use is available to download from [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).
- The [Google dataset](https://www.kaggle.com/lava18/google-play-store-apps) we use is available to download from [this link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).

In [1]:
#Apple Dataset Importing
opened_file = open('AppleStore.csv', encoding="utf8")
from csv import reader
read_file = reader(opened_file)
apple = list(read_file)

#Google Dataset Importing
opened_file = open('googleplaystore.csv', encoding="utf8")
read_file = reader(opened_file)
google = list(read_file)

#We separate the header and data values for ease later on
apple_header = apple[0]
apple = apple[1:]
google_header = google[0]
google = google [1:]

# Exploring the Data

In [2]:
#Explore Data function to let us examine the data, see some of the values etc.
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n')
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
        
#Exploring the Data in the Dataset, to see if this is what we expect
explore_data(apple, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


We can see that in the Apple App Store Dataset, there are many columns of data available to use. In this project, the most relevant ones are: `track_name` , `currency`, `price`, `rating_count_tot`, `rating_count_ver`, and `prime_genre`. 

Each column title can be reviewed for understanding from this [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home).

The following shows data from the Google Play Market. 

In [3]:
explore_data(google, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


The columns of data that are useful for our analysis are: `App`, `Category`, `Reviews`, `Installs`, `Type`, `Price`, and `Genres`. 

Each column title can be reviewed for understanding from this [documentation](https://www.kaggle.com/lava18/google-play-store-apps).

# Deleting Incomplete Entries

In [4]:
for row in apple:
    headerlength = len(apple_header)
    rowlength = len(row)
    if rowlength != headerlength:
        #These commented sections are used to see what data we are about to remove, and to ensure they are in fact errored.
        #print(row)
        #print(apple.index(row))
        del apple[apple.index(row)]
        
for row in google:
    headerlength = len(google_header)
    rowlength = len(row)
    if rowlength != headerlength:
        #print(row)
        #print(google.index(row))
        #The line below is for double checking and ensuring this line was the one indicated by the code above.        
        #print(google[10472])
        del google[google.index(row)]

We can see that there is an error in the Google Play Store dataset. The row length isn't what we expect it to be, indicating there is missing data. As a result, we must delete this line, for it to not affect our analysis.

By using the `del` function, in the context of `google` or `apple` (our dataset variables), we can combine our scanning loop with a delete function, to also remove the incorrect data.

# Managing Duplicate Entries
We don't want any duplicate entries in the dataset, as it will skew the data analysis. For instance, in the Google Play Market dataset, we can see there are duplicates present for 'Instagram'.

In [5]:
for app in google:
    name = app[0]
    if name == 'Instagram':
        print(app)
# for app in apple:
#     name = app[1]
#     if name == 'Mannequin Challenge' or name == 'VR Roller Coaster':
#         print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In the above example, we can see there are 4 entries for Instagram present. Using the code below, we can see the number of duplicate entries in either dataset:

In [6]:
#Apple
duplicates = []
uniques = []
for app in apple:
    name = app[1]
    if name in uniques:
        duplicates.append(name)
    else:
        uniques.append(name)
print('The number of Duplicate Entries in the Apple Dataset:', len(duplicates))
print('The number of Unique Entries in the Apple Dataset:', len(uniques))
#print(duplicates)
print("")

#Google
duplicates = []
uniques = []
for app in google:
    name = app[0]
    if name in uniques:
        duplicates.append(name)
    else:
        uniques.append(name)
print('The number of Duplicate Entries in the Google Dataset: ', len(duplicates))
print('The number of Unique Entries in the Google Dataset: ', len(uniques))
#print(duplicates)

The number of Duplicate Entries in the Apple Dataset: 2
The number of Unique Entries in the Apple Dataset: 7195

The number of Duplicate Entries in the Google Dataset:  1181
The number of Unique Entries in the Google Dataset:  9659


We do not want to remove our data randomly. Instead we will use a specfic criterion to decide which data remains. Below is one of the data entries in question:

`['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']`

In this, the *66577313* refers to the number of reviews the app has, at the moment it was scraped. As a result, we can assume the entry with the most number of reviews is the most recent. We will use this technique to determine which data to keep.

In [7]:
reviews_max = {}
for app in google:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
#We only need to run this code for the google dataset, as we know there is no duplicate entries present in the apple dataset
#We know there were 9659 unique entires from before, and we want ot make sure that there is 9659 entries in the dictionary.
print(len(reviews_max))

9659


As seen, we got a 9659 total values in the table, which is what we were aiming for. We can now use `reviews_max` to create a new dataset, but with clean entries. This is seen below:

In [8]:
google_clean = []
already_added = []
for app in google:
    name = app[0]
    n_reviews = float(app[3])
    if (name not in already_added) and (reviews_max[name] == n_reviews):
        google_clean.append(app)
        already_added.append(name)
#This is to check we have got the correct length
explore_data(google_clean, 0, 2, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 9659
Number of columns: 13


By adding all the clean data to a dictionary, which we use as a reference to create a new list, we now have the correct length of new clean data. By using the `explore_data` function we created earleir, we are able to to comb through our data and see if it is of the expected length.

By using the combined `if` statements earlier, we are then able to select only apps with the highest number of reviews and add them to the list.

Note: The following is a process to clean the apple data set using the same method as listed above.

In [9]:
#Resetting Variables to allow us to clean the Apple Dataset.
reviews_max = {}
for app in apple:
    name = app[1]
    n_reviews = float(app[6])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
apple_clean = []
already_added = []
for app in apple:
    name = app[1]
    n_reviews = float(app[6])
    if (name not in already_added) and (reviews_max[name] == n_reviews):
        apple_clean.append(app)
        already_added.append(name)
#This is to check we have got the correct length
explore_data(apple_clean, 0, 2, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


Number of rows: 7195
Number of columns: 16


# Removing Non-English Applications

Our company focuses on producing apps for the englishs-speaking market, as a result, we aim to produce apps that will be popular in those markets; we need to remove non-english applications from both datasets.

In order to achieve our goals of scanning both databases, we can create a universal function that will check if the app is in English. Too limit the effect of data-loss, we will only remove the entry, if there are more than 3 non-ASCII english characters.

In [10]:
def check_english(phrase):
    non_english_count = 0
    for char in phrase:
        if ord(char) > 127:
            non_english_count += 1
    #This count only allows the app to fail the check if there are more than 3 non-standard characters. This allows for non
    #standard characters in some English app titles, such as Emojis.
    if non_english_count > 3:
        return False
    return True

In [11]:
#Apple (English Apps Only)
apple_english = []
for app in apple_clean:
    name = app[1]
    if check_english(name):
        apple_english.append(app)
#print(len(apple_english))
        
#Google (English Apps Only)
google_english = []
for app in google_clean:
    name = app[0]
    if check_english(name):
        google_english.append(app)
#print(len(google_english))

print('English Apple Dataset')
explore_data(apple_english, 0, 0, True)
print('')
print('English Google Dataset')
explore_data(google_english, 0, 0, True)

English Apple Dataset
Number of rows: 6181
Number of columns: 16

English Google Dataset
Number of rows: 9614
Number of columns: 13


We can see in the new datset, the English-only component contains only 6181 entries in the Apple App Store, and 9614 entries in the Google Play Markets.

# Isolating the Free Apps

Since our company revolves around the use of free apps, we want to purge all non-free apps from our dataset to allow for targetted analysis.

In [12]:
# print(apple_header)
# We can, price is in the 5th Column
# print(google_header)
# We can, price is in the 8th Column
final_apple = []
final_google = []

# Apple Dataset
for app in apple_english:
    price = app[4]
    if price == '0.0':
        final_apple.append(app)

# Google Dataset
for app in google_english:
    price = app[7]
    if price == '0':
        final_google.append(app)
        
print('The number of free apps in the Apple dataset is', len(final_apple))
print('The number of free apps in the Google dataset is', len(final_google))

The number of free apps in the Apple dataset is 3220
The number of free apps in the Google dataset is 8864


We can see we are left with 3220 apps from the Apple dataset, and 8864 apps from the Google dataset. This is what is used for analysis.
# Filtering Apps by Genre
To minimize risks and overhead, our validation strategy for an app idea is comprised of three step:
1. Build a minimal Android version of the app, and add it to Google Play.
1. If the app has a good response from users, we then develop it further.
1. If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.

Since our end goal is to produce an app that is suitable for both markets, our aim is to find a profile suitable for both demographics. We can start to do this by looking at the most commmon category of apps in either market. For this, we'll build a frequency table for the `prime_genre` column of the App Store data set, and the `Genres` and `Category` columns of the Google Play data set.

We can do this by creating frequency tables, we can use two different functions to analyze either frequency table to help us focus our aims:
- One function to generate frequency tables that show percentages
- Another function we can use to display the percentages in a descending order

In [13]:
def freq_table(dataset, index):
    table = {}
    total = 0
    #This for loop will create a frequency table with numbers
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
 
    #This section of the code focuses on converting our raw data into percentages
    percentage_table = {}
    for key in table:
        percentage = (table[key] / total * 100)
        percentage_table[key] = percentage    
    return percentage_table

#<------------------------------------------------------>
#This code below was provided by DataQuest
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
#<------------------------------------------------------>

Having created our function, we can use it to analyse the Apple Appstore, and Google Play Markets data.

### `prime_genre` data in the Apple Appstore Dataset

In [14]:
display_table(final_apple, 11)

Games : 58.13664596273293
Entertainment : 7.888198757763975
Photo & Video : 4.968944099378882
Education : 3.6645962732919255
Social Networking : 3.291925465838509
Shopping : 2.608695652173913
Utilities : 2.515527950310559
Sports : 2.142857142857143
Music : 2.049689440993789
Health & Fitness : 2.018633540372671
Productivity : 1.7391304347826086
Lifestyle : 1.5838509316770186
News : 1.3354037267080745
Travel : 1.2422360248447204
Finance : 1.1180124223602486
Weather : 0.8695652173913043
Food & Drink : 0.8074534161490683
Reference : 0.5590062111801243
Business : 0.5279503105590062
Book : 0.43478260869565216
Navigation : 0.18633540372670807
Medical : 0.18633540372670807
Catalogs : 0.12422360248447205


#### Preliminary Analysis of Data
We can see that the `prime_genre`secion in the Apple Appstore dataset shows that the most common free app available is games, with a 58.13% market share. The follwing apps don't have more than a 10% market share. The second most prevalent category is Entertainment, with a share of 7.89%.

We can see that there is very many more apps designed for entertainment purposes (games, photo and video, social networking, sports, music, etc.), than practical (education, shopping, utilities, productivity, lifestyle, etc.) purposes. However, this does not imply that one category is more popular with users. The demand of apps does not necessarily correlate with the supply of apps.
### `Categories` data in the Google Play Markets Dataset

In [15]:
display_table(final_google, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

#### Preliminary Analysis of Data
We can see that the `Genres`secion of the Google Play Markets Dataset. Immediately, we can see there is a major difference, in terms of apps available on the market. Games is no longer a dominant app style on the market, and there are many diverse types of apps available. The most common being Family apps _(which comprises of mainly Games)_ with a presence of 18.91%. The second most common is Game, with a presence of 9.72%.

We can also see that there are many more practical-use apps present in the Google Play dataset, suggesting there is a bigger spread of apps available.
### `Genres` data in the Google Play Markets Dataset

In [16]:
display_table(final_google, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

#### Preliminary Analysis of Data
The differences between the `Genres` and `Category` section isn't clear, however, there are many more subdivisons in the `Genres`. As we are looking at an overall picture, we will focus on the `Category` column.
# The Most Popular Apps on the Appstore by Genre

In [17]:
apple_genres = freq_table(final_apple, -5)
apple_genres_sorted = {}
for genre in apple_genres:
    total = 0
    len_genre = 0
    for app in final_apple:
        genre_app = app[-5]
        if genre_app == genre:
            total += float(app[5])
            len_genre += 1
    avg_n_ratings = total / len_genre
    avg_n_ratings = round(avg_n_ratings, 2)
    apple_genres_sorted[genre] = avg_n_ratings
    #print(genre, ':', avg_n_ratings)
    #This creates a list, but it is not easy to read as it is not sorted in any order.
    
#This next section is an adaption of the code provided by DataQuest. This will organise the data in descending order.    
table_display = []
for key in apple_genres_sorted:
    key_val_as_tuple = (apple_genres_sorted[key], key)
    table_display.append(key_val_as_tuple)
    table_sorted = sorted(table_display, reverse = True)
    
for entry in table_sorted:
    print(entry[1], ':', entry[0])

Navigation : 86090.33
Reference : 74942.11
Social Networking : 71548.35
Music : 57326.53
Weather : 52279.89
Book : 39758.5
Food & Drink : 33333.92
Finance : 31467.94
Photo & Video : 28441.54
Travel : 28243.8
Shopping : 26919.69
Health & Fitness : 23298.02
Sports : 23008.9
Games : 22812.92
News : 21248.02
Productivity : 21028.41
Utilities : 18684.46
Lifestyle : 16485.76
Entertainment : 14029.83
Business : 7491.12
Education : 7003.98
Catalogs : 4004.0
Medical : 612.0


On average, we can see that Navigation apps have a very high representation. If we investigate further, we see, due to the very small sample size, this result is skewed:

In [18]:
for app in final_apple:
    if app[-5] == 'Navigation':
        print(app[1], ':', app[5], 'users')

Waze - GPS Navigation, Maps & Real-time Traffic : 345046 users
Google Maps - Navigation & Transit : 154911 users
Geocaching® : 12811 users
CoPilot GPS – Car Navigation & Offline Maps : 3582 users
ImmobilienScout24: Real Estate Search in Germany : 187 users
Railway Route Search : 5 users


If we investigate further, we can see the same theme appararent in Reference and Social Networking:

In [19]:
print('Reference')
count = 0
for app in final_apple:
    if app[-5] == 'Reference' and count < 10:
        print(app[1], ':', app[5], 'users')
        count += 1
print()
count = 0
print('Social Networking')
for app in final_apple:
    if app[-5] == 'Social Networking' and count < 10:
        print(app[1], ':', app[5], 'users')
        count += 1
print()
count = 0
print('Music')
for app in final_apple:
    if app[-5] == 'Music' and count < 10:
        print(app[1], ':', app[5], 'users')
        count += 1

Reference
Bible : 985920 users
Dictionary.com Dictionary & Thesaurus : 200047 users
Dictionary.com Dictionary & Thesaurus for iPad : 54175 users
Google Translate : 26786 users
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418 users
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588 users
Merriam-Webster Dictionary : 16849 users
Night Sky : 12122 users
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535 users
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693 users

Social Networking
Facebook : 2974676 users
Pinterest : 1061624 users
Skype for iPhone : 373519 users
Messenger : 351466 users
Tumblr : 334293 users
WhatsApp Messenger : 287589 users
Kik : 260965 users
ooVoo – Free Video Call, Text and Voice : 177501 users
TextNow - Unlimited Text + Calls : 164963 users
Viber Messenger – Text & Call : 164249 users

Music
Pandora - Music & Radio : 1126879 users
Spotify Music : 87

Taking a look at our data, we see this similar trend continue. However, given how saturated the Apple App Store is with Gaming Applications, it would be unwise to develop in that direction. As a result, it's better to pick something out of these categories. Creating an application for the Reference category appears to be our best bet. Converting a book to an application, it isn't a highly contested marketspace.

Other popular app categories are outside of our scope as a company and what we can achieve:
- Finance: We are not a financial institute, hence have no reason to manage or restrict a person's money, nor do we have the expertise.
- Weather: We do not have access to meteorologists within our company, so no sense in developing a weather app.
- Food and Drink: We do not have a foodstuff service, therefore, we would have to redirect users elsewhere, which is not our aim, with this app.
Next, we will shift our attention to the Google Play Markets.
# The Most Popular Apps on Google Play Markets by Genre

We can see in this dataset, there is `installs` data, however, as seen here, it can be very imprecise. There is no datapoint,in between values. We do not know if there is 1 million, or 1.2 million for an app which is 1,000,000+.

In [20]:
display_table(final_google, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


In order to get around this issue, we will assume that for an app which is 1,000,000+, it has only 1 million installs and so on. We will do this for each checkpoint value. In order to manage this computationally, we must convert all the installation data entries into `floats`.

In [21]:
categories_google = freq_table(final_google, 1)
google_categories_sorted = {}
for category in categories_google:
    total = 0
    len_category = 0
    for app in final_google:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    avg_n_installs = round(avg_n_installs,2)
    google_categories_sorted[category] = avg_n_installs
table_display = []
for key in google_categories_sorted:
    key_val_as_tuple = (google_categories_sorted[key], key)
    table_display.append(key_val_as_tuple)
    table_sorted = sorted(table_display, reverse = True)
    
for entry in table_sorted:
    print(entry[1], ':', entry[0])

COMMUNICATION : 38456119.17
VIDEO_PLAYERS : 24727872.45
SOCIAL : 23253652.13
PHOTOGRAPHY : 17840110.4
PRODUCTIVITY : 16787331.34
GAME : 15588015.6
TRAVEL_AND_LOCAL : 13984077.71
ENTERTAINMENT : 11640705.88
TOOLS : 10801391.3
NEWS_AND_MAGAZINES : 9549178.47
BOOKS_AND_REFERENCE : 8767811.89
SHOPPING : 7036877.31
PERSONALIZATION : 5201482.61
WEATHER : 5074486.2
HEALTH_AND_FITNESS : 4188821.99
MAPS_AND_NAVIGATION : 4056941.77
FAMILY : 3695641.82
SPORTS : 3638640.14
ART_AND_DESIGN : 1986335.09
FOOD_AND_DRINK : 1924897.74
EDUCATION : 1833495.15
BUSINESS : 1712290.15
LIFESTYLE : 1437816.27
FINANCE : 1387692.48
HOUSE_AND_HOME : 1331540.56
DATING : 854028.83
COMICS : 817657.27
AUTO_AND_VEHICLES : 647317.82
LIBRARIES_AND_DEMO : 638503.73
PARENTING : 542603.62
BEAUTY : 513151.89
EVENTS : 253542.22
MEDICAL : 120550.62


We can see that communication contains apps, which have the most number of installs, however, this is a heavily skewed result. There are some apps with over a billion installs, with a few more with 100 million+:

In [22]:
for app in final_google:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+' or app[5] == '500,000,000+' or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

By removing all these apps, we can see a drastic reduction in the number of installs of apps from this category:

In [23]:
under_100m = []
for app in final_google:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100m.append(float(n_installs))
print(round((sum(under_100m) / len(under_100m)),2))

3603485.39


A similar pattern can be seen with the video players, social, photography, and productivity categories:

In [24]:
#Video Players
under_100m = []
for app in final_google:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'VIDEO_PLAYERS') and (float(n_installs) < 100000000):
        under_100m.append(float(n_installs))
print("VIDEO_PLAYERS :",round((sum(under_100m) / len(under_100m)),2))

#Social
under_100m = []
for app in final_google:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'SOCIAL') and (float(n_installs) < 100000000):
        under_100m.append(float(n_installs))
print("SOCIAL :",round((sum(under_100m) / len(under_100m)),2))

#Photography
under_100m = []
for app in final_google:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'PHOTOGRAPHY') and (float(n_installs) < 100000000):
        under_100m.append(float(n_installs))
print("PHOTOGRAPHY :",round((sum(under_100m) / len(under_100m)),2))

#Productivty
under_100m = []
for app in final_google:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'PRODUCTIVITY') and (float(n_installs) < 100000000):
        under_100m.append(float(n_installs))
print("PRODUCTIVITY :",round((sum(under_100m) / len(under_100m)),2))

VIDEO_PLAYERS : 5544878.13
SOCIAL : 3084582.52
PHOTOGRAPHY : 7670532.29
PRODUCTIVITY : 3379657.32


We already know the game genre is rather saturated, so we'd be better off targetting a different category. We previously discussed the Reference category, and even on the Google Play Markets, it is a fairly popular category, with around 8.8 million installs per app. If we remove any extreme apps, we get:

In [25]:
#Books and Reference
under_100m = []
for app in final_google:
    n_installs = app[5]
    n_installs = n_installs.replace(',', '')
    n_installs = n_installs.replace('+', '')
    if (app[1] == 'BOOKS_AND_REFERENCE') and (float(n_installs) < 100000000):
        under_100m.append(float(n_installs))
print("BOOKS_AND_REFERENCE :",round((sum(under_100m) / len(under_100m)),2))

BOOKS_AND_REFERENCE : 1437212.22


There are still more than 1 million downloads per app. This shows it can be a fairly popular category.
# Conclusion
We should advise our developers to produce an app which focuses on converting a popular/ recent book, into an app. It is niche, but unsaturated market, meaning an app that we develop will stand out. This in turn can be highly profitable in both markets.